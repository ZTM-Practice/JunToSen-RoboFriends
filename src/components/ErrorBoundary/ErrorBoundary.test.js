import { shallow } from "enzyme";
import React from "react";
import ErrorBoundary from "./ErrorBoundary";

describe('ErrorBoundary Component', () => {
    it('renders ErrorBoundary Component', () => {
        expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
    });

    it('displays error message if hasError is true', () => {
        const wrapper = shallow(<ErrorBoundary />);
        const instance = wrapper.instance();
        instance.state = { hasError: true };
        wrapper.setProps({})
        expect(wrapper.exists('#errorMessage')).toEqual(true);
    });

    it('doesn\'t display error message if hasError is false', () => {
        const wrapper = shallow(<ErrorBoundary />);
        expect(wrapper.exists('#errorMessage')).toEqual(false);
    })
});