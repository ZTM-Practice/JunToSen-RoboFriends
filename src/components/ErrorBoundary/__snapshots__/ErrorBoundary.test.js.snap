// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component renders ErrorBoundary Component 1`] = `ShallowWrapper {}`;
